src/map/script.c

```diff
@@ -24559,6 +24559,26 @@ static BUILDIN(getcalendartime)
 	return true;
 }
 
+static BUILDIN(setfavoriteitem)
+{
+	struct map_session_data *sd = script_rid2sd(st);
+	int idx = script_getnum(st, 2);
+	int value = script_getnum(st, 3);
+
+	if (sd == NULL) {
+		ShowError("buildin_setfavoriteitem: no player attached.\n");
+		return false;
+	} else if (idx < 0 || sd->status.inventorySize <= 0 || idx >= sd->status.inventorySize) {
```

Comment:
what's the point of having `sd->status.inventorySize <= 0` ?
isn't this always between 100 and MAX_INVENTORY ?

Reply:
checking condition for if player has no item in inventory

Reply:
no, `sd->status.inventorySize` is equivalent to `*getInventorySize` script command
`sd->status.inventorySize` is the maximum amount of different items the player can hold

for example, if I have 10 items inside my inventory, 
`sd->status.inventorySize` is still 100, and can only be expand with `*expandInventory` script command

Reply:
i see. thx for the info. updated it.